src/differentials/abstract_zero.jl

```diff
@@ -24,6 +24,8 @@ Base.adjoint(z::AbstractZero) = z
 Base.transpose(z::AbstractZero) = z
 Base.:/(z::AbstractZero, ::Any) = z
 
+Base.convert(::Type{T}, x::AbstractZero) where T <: Real = zero(T)
```

Comment:
Any reason to restrict this to `Real`? I think it would make sense to at least allow `Complex`, but all subtypes of `Number` should work as well here.

Reply:
Oh, and I don't know if we really want to implement this for all `AbstractZero`s. I am not sure this makes sense for `DoesNotExist()`, so it might be better to do this just for `Zero`.

Reply:
Thanks for catching the `Real` -> `Number`, it was just an error on my part.

My understanding of the `Zero` vs `DoesNotExist` differential types is that the purpose of the distinction between them is primarily to signal meaning when defining rules. We do define mathematical operations on `DoesNotExist` in which it behaves like a zero, so it seems we aren't introducing any `ideologically new` behaviour by allowing it to be converted to a zero?

Reply:
```suggestion
Base.convert(::Type{T}, x::AbstractZero) where T <: Number = zero(T)
```